### PR TITLE
chore(dashpay): restoring Contacts flow

### PR DESCRIPTION
**Pull request by Syn-McJ**
## Issue being fixed or feature implemented
As a part of the "Bringing back DashPay" initiative, the contacts entry point in the tabbar should be restored.

## What was done?
- Fix some build errors
- Restore Contacts and Explore tabs in the tabbar
- Update the tabbar icons (selected/unselected) 
- Mock the username in `dashPayModel` to sync with the `HomeHeaderView`
- Mock Contacts when searching
- Use a single flag for DashPay mocking functionality

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: DashPay/Presentation/Home/DWDashPayReadyProtocol.h [hunk 1]
```diff
@@ -24,7 +24,6 @@ extern NSNotificationName const DWDashPayAvailabilityStatusUpdatedNotification;
 @protocol DWDashPayReadyProtocol <NSObject>
 
 @property (readonly, nonatomic, assign) BOOL isDashPayReady;
-@property (readonly, nonatomic, assign) BOOL isDashPayReadyMainSuggestion;
```
Comment (Syn-McJ) [inline DashPay/Presentation/Home/DWDashPayReadyProtocol.h:27]: Not needed atm. Removed to keep the `DWDashPayReadyProtocol` footprint low.

File: DashWallet/Sources/UI/Home/Protocols/DWHomeProtocol.h [hunk 2]
```diff
@@ -42,7 +43,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)homeModelWantToReloadShortcuts:(id<DWHomeProtocol>)model;
 @end
 
-@protocol DWHomeProtocol <DWTxDisplayModeProtocol>
+@protocol DWHomeProtocol <DWTxDisplayModeProtocol, DWDashPayReadyProtocol>
```
Comment (Syn-McJ) [inline DashWallet/Sources/UI/Home/Protocols/DWHomeProtocol.h:46]: Contacts VC requires `DWDashPayReadyProtocol` which was implemented by the home model before. Perhaps there is a path to refactor this in the future stories.

File: DashWallet/Sources/UI/DashPay/DWDashPayConstants.m [hunk 3]
```diff
@@ -24,5 +24,6 @@
 
 NSInteger DW_MIN_USERNAME_LENGTH = 3;
 NSInteger DW_MAX_USERNAME_LENGTH = 24;
+BOOL MOCK_DASHPAY = YES; // TODO: remove once Platform is available
```
Comment (Syn-McJ) [inline DashWallet/Sources/UI/DashPay/DWDashPayConstants.m:27]: I figure it's better to reference a single flag for any mocking code. Will be easier to turn off and remove once the platform is available.

**Review by tikhop — APPROVED** [review]
Looks good!

**Pull request merged**